catalog/transformer/catalog-transformer-html/src/test/java/ddf/catalog/transformer/html/HtmlMetacardTransformerTest.java

```diff
@@ -0,0 +1,191 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.html;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transformer.html.models.HtmlCategoryModel;
+import ddf.catalog.transformer.html.models.HtmlMetacardModel;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HtmlMetacardTransformerTest {
+
+  private static final String METACARD_CLASS = ".metacard";
+  private static final String CATEGORY_CLASS = ".metacard-category";
+  private static final String CATEGORY_TABLE_CLASS = ".category-table";
+  private static final String METACARD_ATTRIBUTE_CLASS = ".metacard-attribute";
+  private static final String EMPTY_ATTRIBUTE_CLASS = ".empty-attribute";
+  private static final String MEDIA_ATTRIBUTE_CLASS = ".media-attribute";
+
+  private static final List<String> EMPTY_ATTRIBUTE_LIST = Collections.emptyList();
+  private static final List<HtmlCategoryModel> EMPTY_CATEGORY_LIST = Collections.emptyList();
+
+  private static final String[] ALL_CATEGORY_TITLES =
+      new String[] {
+        "Associations",
+        "Contact",
+        "Core",
+        "DateTime",
+        "Location",
+        "Media",
+        "Security",
+        "Topic",
+        "Validation",
+        "Version"
+      };
+
+  private List<String> associationsList;
+  private List<String> coreList;
+
+  @Before
+  public void setup() {
+    associationsList = new ArrayList<>();
+    associationsList.add(Associations.RELATED);
+    associationsList.add(Associations.DERIVED);
+    associationsList.add(Associations.EXTERNAL);
+
+    coreList = new ArrayList<>();
+    coreList.add(Core.THUMBNAIL);
+  }
+
+  @Test
+  public void testMetacardCreation() {
+    Metacard metacard = new MetacardImpl();
+
+    List<HtmlMetacardModel> metacardModelList = new ArrayList<>();
+    metacardModelList.add(new HtmlMetacardModel(metacard, EMPTY_CATEGORY_LIST));
+    metacardModelList.add(new HtmlMetacardModel(metacard, EMPTY_CATEGORY_LIST));
+
+    HtmlMetacardTransformer htmlTransformer = new HtmlMetacardTransformer();
+
+    Document doc = Jsoup.parse(htmlTransformer.buildHtml(metacardModelList));
+
+    assertThat(doc.select(METACARD_CLASS).size(), is(metacardModelList.size()));
```

Comment:
Use the `hasSize()` matcher for better failure messages and to make the intent of your test clearer.